common/lcs/src/ser.rs

```diff
@@ -73,6 +73,15 @@ impl Serializer {
     fn end(self) -> Vec<u8> {
         self.output
     }
+
+    /// Serialize a sequence length as a u32.
+    fn serialize_seq_len(&mut self, len: usize) -> Result<()> {
+        use serde::ser::Serializer;
+        if len > crate::MAX_SEQUENCE_LENGTH {
+            return Err(Error::ExceededMaxLen(len));
+        }
+        self.serialize_u32(len as u32)
+    }
```

Comment:
Can you move this to be closer to the `serialize_seq` function? Like right above it?

Reply:
`serialize_seq` is part of the `serde::ser::Serializer` trait, so I can't move `serialize_seq_len` above it, as it's not part of the trait. :(